packages/theme/src/cli/utilities/theme-environment/html.ts

```diff
@@ -98,3 +102,20 @@ function getErrorPage(options: {title: string; header: string; message: string;
     </body>
   </html>`
 }
+
+function assertThemeId(response: Response, html: string, expectedThemeId: string) {
+  const obtainedThemeId = html.match(/Shopify\.theme\s*=\s*{[^}]+?"id":\s*"?(\d+)"?(}|,)/)?.[1]
```

Comment:
I wonder if it's worth adding a comment above this to explain what we're matching against? Something like:

```js
// Match against the `Shopify.theme` object that is defined in the markup like this:
// Shopify.theme = {"name":"Dawn","id":1234,"theme_store_id":null,"role":"development"};
```